integration-tests/api/__tests__/admin/product.js

```diff
@@ -2585,102 +2546,6 @@ medusaIntegrationTestRunner({
           expect(variantPost).not.toBeTruthy()
         })
 
-        // TODO: This one is a bit more complex, leaving for later
-        it.skip("successfully deletes a product variant and its associated option values", async () => {
```

Comment:
This is testing implementation detail of the module, and as such I moved the tests to the module (there already was a test for the product, I just added one for the variant)